frontend/packages/knative-plugin/src/components/add/EventSourcePage.tsx

```diff
@@ -4,25 +4,53 @@ import { RouteComponentProps } from 'react-router';
 import { useTranslation } from 'react-i18next';
 import { PageBody } from '@console/shared';
 import { LoadingInline, PageHeading } from '@console/internal/components/utils';
+import { K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
 import NamespacedPage, {
   NamespacedPageVariants,
 } from '@console/dev-console/src/components/NamespacedPage';
 import { QUERY_PROPERTIES } from '@console/dev-console/src/const';
 import ConnectedEventSource from './EventSource';
 import EventSourceAlert from './EventSourceAlert';
-import { useEventSourceList } from '../../utils/create-eventsources-utils';
+import {
+  addCamelKameletSourceList,
+  useEventSourceList,
+} from '../../utils/create-eventsources-utils';
 import { isDynamicEventSourceKind } from '../../utils/fetch-dynamic-eventsources-utils';
+import { CamelKameletModel } from '../../models';
 
 type EventSourcePageProps = RouteComponentProps<{ ns?: string }>;
 
 const EventSourcePage: React.FC<EventSourcePageProps> = ({ match, location }) => {
   const { t } = useTranslation();
   const namespace = match.params.ns;
-  const eventSourceStatus = useEventSourceList(namespace);
+  const eventSourceList = useEventSourceList(namespace);
   const searchParams = new URLSearchParams(location.search);
   const sourceKindProp = searchParams.get('sourceKind');
-  const isSourceKindPresent = sourceKindProp && isDynamicEventSourceKind(sourceKindProp);
+  const kameletName = sourceKindProp && searchParams.get('name');
+  const watchResource = React.useMemo(
+    () => ({
+      [CamelKameletModel.plural]: {
+        kind: referenceForModel(CamelKameletModel),
+        name: kameletName,
+        namespace,
+        optional: true,
+        selector: {
+          matchLabels: {
+            'camel.apache.org/kamelet.type': 'source',
+          },
+        },
+      },
+    }),
+    [kameletName, namespace],
+  );
+  const {
+    kamelets: { data: kamelet, loaded: kameletLoaded },
+  } = useK8sWatchResources<{ [key: string]: K8sResourceKind }>(watchResource);
```

Comment:
missed this , would be better to use `useK8sWatchResource`

Reply:
used `useK8sGet` as we no need to watch on the resource.